homeassistant/components/google_assistant/trait.py

```diff
@@ -196,16 +197,22 @@ def _next_selected(items: list[str], selected: str | None) -> str | None:
     return items[next_item]
 
 
-class _Trait:
+class _Trait(ABC):
     """Represents a Trait inside Google Assistant skill."""
 
+    name: str
     commands: list[str] = []
 
     @staticmethod
     def might_2fa(domain, features, device_class):
         """Return if the trait might ask for 2FA."""
         return False
 
+    @staticmethod
+    @abstractmethod
+    def supported(domain, features, device_class, attributes):
```

Comment:
Missing type hints, is that for a reason?

Reply:
This PR adds strict type checks only of the `cloud` integration.

Reply:
That is clear, I only ask this as you are changing code. It might trigger other issues in this case.

Reply:
This is added because not having the `supported` method in the base class makes mypy angry. Typing of google_assistant can be done in a future PR 👍